### ISSUE TITLE
Autocomplete: Listbox missing accessible name

### ISSUE THREAD

**Issue by selfthinker** [alphagov/govuk-design-system#4005]
This issue is from the accessibility audit of the Design System website by DAC in July 2024.

## DAC's description

* Page: [Home – GOV.UK Design System](https://design-system.service.gov.uk/)
* WCAG reference: 4.1.2 Name, Role, Value (Level A)
* Issue ID: DAC_Autocomplete_Semantics_01 (same ID used for #4004)

Also, the listbox that contains all the relevant options does not have an accessible name.
When a user navigates to a listbox using a screen reader, the screen reader will announce the accessible name of the control, followed by the options within the list. If the accessible name is not provided, then screen reader users might not be able to understand the content and what is represents.

<img width="691" alt="Screenshot of the autocomplete in action" src="https://github.com/user-attachments/assets/5f5b3fb4-25fa-4734-987b-5df0c6debed7">

### Current code
```html
<ul id="app-site-search__input__listbox" role="listbox" class="app-site-search__menu app-site-search__menu--overlay app-site-search__menu--visible">
  <li aria-selected="false" class="app-site-search__option app-site-search__option--focused" id="app-site-search__input__option--0" role="option" tabindex="-1" aria-posinset="1" aria-setsize="1">
    Radios<span class="app-site-search--section">Components</span>
  </li>
</ul>
```

## DAC's proposed solution

Also, provide the listbox with an accessible name. This can either by done via the `aria-label` or `aria-labelledby` attribute.

## Additional instances

This component appears on every page but is a single component, therefore only needs to be fixed once.
It uses the [accessible autocomplete component](https://github.com/alphagov/accessible-autocomplete).

## Needed roles

Frontend developer
